documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -0,0 +1,263 @@
+
+## Overview
+
+<!-- An overview of the implementation plan, if necessary. Save any specific steps for the section(s) below. -->
+
+This plan describes the changes that need to be added on the frontend: the new
+components, pages and store.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+Three collection pages are added to Openverse:
+
+- items with the selected tag
+- items from the selected source (provider)
+- items by the selected creator. The pages re-use the existing image grid and
+  audio collection views and can load more images or audio items on the Load
+  more button click. The single result pages are updated to add the links to
+  these pages, as seen in the Figma mockups: [link to Figma mockup]
+
+## Outlined Steps
+
+<!-- Describe the implementation step necessary for completion. -->
+
+### Page changes
+
+#### Create a page for `tag` / `creator` /`source` collections.
+
+Nuxt allows creating nested dynamic routes like
+`/pages/_collection/_mediaType/_term`.
+
+Here, the collection can be `tag`, `creator` or `source`. `mediaType` can be one
+of the supported media types (currently, `image` and `audio`). `term` refers to
+the actual value of the tag, creator or source name.
+
+This page should use the
+[`validate` method](https://v2.nuxt.com/docs/components-glossary/validate/) to
+make sure that the `collection`, `searchType` and `term` `params` are valid.
+
+```
+`validate({ params, $pinia }): {
+const { collection, searchType, term } = params
+```
+
+This page should also update the state (`searchType`, `searchTerm` and
+`searchBy` filter) in the `search` store and handle fetching using
+`mediaStore`'s `fetchMedia` method in the `useFetch` hook. Since it is not
+possible to change the path or query parameters from this page client-side,
+fetching can be much simpler than on the current search page (that has to watch
+for changes in the route and fetch if necessary).
+
+This page should have the common header (title, button - link to the source
+external URL or the creator URL, and the summary of item counts in the
+collection), and a nested component: the image grid or the audio collection.
+
+### New and updated components
+
+#### Extract the `VAudioCollection` component
+
+Currently, it is not possible to reuse the audio collection from the audio
+search result page because it is a part of the `audio.vue` page. We should
+extract the part that shows the loading skeleton, the column of `VAudioTrack`
+rows and the Load more section. This component will be reused in the audio
+search page and on the Additional search views. One thing that I'm not sure
+about here is whether the analytics events should stay the same and be sent from
+all pages - or if we should somehow differentiate the events coming from
+different search views.
```

Comment:
> One thing that I'm not sure
about here is whether the analytics events should stay the same and be sent from
all pages - or if we should somehow differentiate the events coming from
different search views.

These will be naturally differentiated by the built-in `pathname` property of all analytics events. They can remain the same.